misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/MockAnnotationProcessor.java

```diff
@@ -122,6 +134,95 @@ public void process(SimpleVariableNode simpleVariableNode, List<AnnotationAttach
         }
     }
 
+    @Override
+    public void process(FunctionNode functionNode, List<AnnotationAttachmentNode> annotations) {
+
+        parent = (BLangPackage) ((BLangFunction) functionNode).parent;
+        String packageName = getPackageName(parent);
+        annotations = annotations.stream().distinct().collect(Collectors.toList());
+
+        // Iterate through all the annotations
+        for (AnnotationAttachmentNode attachmentNode : annotations) {
+            String annotationName = attachmentNode.getAnnotationName().getValue();
+            String functionName = functionNode.getName().getValue();
+
+            if (MOCK_ANNOTATION_NAME.equals(annotationName)) {
+                String[] vals = new String[2];
+                // TODO: when default values are supported in annotation struct we can remove this
+                vals[0] = packageName;
+
+                if (attachmentNode.getExpression() instanceof BLangRecordLiteral) {
+                    List<RecordLiteralNode.RecordField> attributes = ((BLangRecordLiteral) attachmentNode
+                            .getExpression()).getFields();
+                    attributes.forEach(field -> {
+                        String name;
+                        BLangExpression valueExpr;
+
+                        if (field.isKeyValueField()) {
+                            BLangRecordLiteral.BLangRecordKeyValueField attributeNode =
+                                    (BLangRecordLiteral.BLangRecordKeyValueField) field;
+                            name = attributeNode.getKey().toString();
+                            valueExpr = attributeNode.getValue();
+                        } else {
+                            BLangRecordLiteral.BLangRecordVarNameField varNameField =
+                                    (BLangRecordLiteral.BLangRecordVarNameField) field;
+                            name = varNameField.variableName.value;
+                            valueExpr = varNameField;
+                        }
+
+                        String value = valueExpr.toString();
+
+                        if (MODULE.equals(name)) {
+                            value = formatPackageName(value, parent);
+                            vals[0] = value;
+                        } else if (FUNCTION.equals(name)) {
+                            vals[1] = value;
+                        }
+                    });
+
+                    // Check if Function in annotation is empty
+                    if (vals[1].isEmpty()) {
```

Comment:
Can this cause NPE  ?

Reply:
I just verified it. Yes. If you provide only the `moduleName` and forget to put the `functionName` in the annotation an NPE is thrown. I will change the implementation.

Reply:
I initialized vals[1] with an empty by default so that the NPE wouldnt arise. 

But ideally, the default value provided in the mock annotation should be applied in this case.